flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemOutputFormat.java

```diff
@@ -81,11 +91,29 @@ private FileSystemOutputFormat(
         this.formatFactory = formatFactory;
         this.computer = computer;
         this.outputFileConfig = outputFileConfig;
+        this.identifier = identifier;
+        this.policyKind = policyKind;
+        this.customClass = customClass;
+        this.successFileName = successFileName;
     }
 
     @Override
     public void finalizeGlobal(int parallelism) {
         try {
+            List<PartitionCommitPolicy> policies =
+                    PartitionCommitPolicy.createPolicyChain(
+                            Thread.currentThread().getContextClassLoader(),
```

Comment:
Use `Thread.currentThread().getContextClassLoader()` to load user class may casue class not found issue.
I think we can make `FileSystemOutputFormat` extend `RichOutputFormat` and use `getRuntimeContext().getUserCodeClassLoader()`.